### PR TITLE
Adding support for retrieving dynamic system properties

### PR DESCRIPTION
**Pull request by Akash-Anand0744**
Adding support for retrieving dynamic system properties i.e., **get_sp_var("flow_input")** where _'flow_input'_ is an input to a flow and returns the defined system property based on user input(_i.e., flow_input_)

### CODE REVIEW

**Review by sashutosh — APPROVED** [review]

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/scripts/ScriptEvaluator.java [hunk 1]
```diff
@@ -80,7 +80,13 @@ private Value doEvaluateExpressionJython(String expr,
                                              Set<SystemProperty> systemProperties,
                                              Set<ScriptFunction> functionDependencies) {
         Map<String, Serializable> jythonContext = createJythonContext(context);
-        boolean systemPropertiesDefined = functionDependencies.contains(ScriptFunction.GET_SYSTEM_PROPERTY);
+
+        boolean systemPropertiesDefined = false;
+        if (functionDependencies.contains(ScriptFunction.GET_SYSTEM_PROPERTY) ||
+                functionDependencies.contains(ScriptFunction.GET_SP_VAR)) {
```
Comment (lucian-cm) [inline cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/scripts/ScriptEvaluator.java:86]: || should be on newline not on first line

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/scripts/ScriptEvaluator.java [hunk 2]
```diff
@@ -93,7 +99,11 @@ private Value doEvaluateExpressionExternalPython(String expr,
                                                      Set<SystemProperty> systemProperties,
                                                      Set<ScriptFunction> functionDependencies) {
         Map<String, Serializable> pythonContext = createExternalPythonContext(context);
-        boolean systemPropertiesDefined = functionDependencies.contains(ScriptFunction.GET_SYSTEM_PROPERTY);
+        boolean systemPropertiesDefined = false;
+        if (functionDependencies.contains(ScriptFunction.GET_SYSTEM_PROPERTY) ||
+                functionDependencies.contains(ScriptFunction.GET_SP_VAR)) {
```
Comment (lucian-cm) [inline cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/scripts/ScriptEvaluator.java:104]: same here

File: cloudslang-runtime/src/main/resources/scripts/get_sp_var.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+def get_sp_var(key):
+  try:
+    value1 = get_from_smaller_context(key)
+  except NameError:
+    value1 = globals().get(key)
+  accessed(value1)
+  property_value = sys_prop.get(value1)
+  return property_value
```
Comment (lucian-cm) [inline cloudslang-runtime/src/main/resources/scripts/get_sp_var.py:8]: you might want to have a new line at the end of the file

**Review by lucian-cm — CHANGES_REQUESTED** [review]

**Comment by lucian-cm** [comment]
@Akash-Anand0744  please add description to the PR and also Signoff comment

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/scripts/ScriptEvaluator.java (hunk 1 above)
Comment (Akash-Anand0744) [reply cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/scripts/ScriptEvaluator.java]: Done

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/scripts/ScriptEvaluator.java (hunk 2 above)
Comment (Akash-Anand0744) [reply cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/scripts/ScriptEvaluator.java]: Done

File: cloudslang-runtime/src/main/resources/scripts/get_sp_var.py (hunk 3 above)
Comment (Akash-Anand0744) [reply cloudslang-runtime/src/main/resources/scripts/get_sp_var.py]: Done

**Review by lucian-cm — APPROVED** [review]

**Comment by sashutosh** [comment]
run-pr-build

**Pull request merged**